### DISCUSSION TITLE
How can I investigate why some command id:s (`execCopy` and `execPaste`) fail for some users

### DISCUSSION THREAD

**Discussion by PEZ** [microsoft/vscode-discussions#235] [Extension Development QnA] [answered]
For my extension [Paste Replaced](https://github.com/PEZ/paste-replaced) I rely on executing the commands `execPaste` and `execCopy`. It works on my machine™️, but some users get the error message that these commands are not available:

* https://github.com/PEZ/paste-replaced/issues/3
* https://github.com/PEZ/paste-replaced/issues/4

I don't understand why these commands would not be available, why they would be available for some users but not others, or how to investigate the issue when I do not have the error myself.

Anyone else have some ideas what I could do about this?

**Comment by gjsjohnmurray** [comment]
Based on https://github.com/PEZ/paste-replaced/issues/4#issuecomment-1313235693 my guess is that your `activate` method is erroring at some point between those two messages it writes.

**Reply by gjsjohnmurray** [reply]
Your .cljs files are a foreign language to me, but if https://github.com/PEZ/paste-replaced/blob/4ca1bf65c899a9819d84b423ad480d369cc82a9e/src/paste_replaced/extension.cljs#L29 is accessing `vscode.workspace.rootPath` (deprecated) perhaps you aren't catering for the case where it returns `undefined`.

**Reply by PEZ** [reply]
Thanks for looking at it, @gjsjohnmurray! I'm never using that rootPath, as it happens, so it is not the cause. And the extension does register the commands (otherwise the `execPaste` would never be attempted). Nonetheless, I've updated the extension to not use the deprecated attribute, and also to print a message in the Output channel if activation fails.

**Reply by PEZ** [reply]
Your comment on the ClojureScript code triggered me. 😄 I now made a video about some of the benefits of using it for developing VS Code Extensions. https://www.youtube.com/watch?v=fMtvtAfdr90

**Comment by PEZ** [comment, marked as answer]
The solution here was to have helpful users. It was a difference between Mac and other platforms around which commands are actually available. My users found that instead of `execPaste`, on their machines the Shortcuts editor showed `editor.action.clipboardPasteAction` as the command ID, and that works on my machine as well. I even got a PR fixing the issue! 🎉 ❤️
